pallets/rmrk-core/src/functions.rs

```diff
@@ -291,20 +304,76 @@ where
 					royalty = Some(RoyaltyInfo::<T::AccountId> { recipient, amount });
 				},
 				None => {
-					royalty =
-						Some(RoyaltyInfo::<T::AccountId> { recipient: owner.clone(), amount });
+					// If a royalty amount is passed but no recipient, defaults to the sender
+					royalty = Some(RoyaltyInfo::<T::AccountId> { recipient: sender, amount });
 				},
 			}
 		};
 
-		let owner_as_maybe_account = AccountIdOrCollectionNftTuple::AccountId(owner.clone());
+		let nft = NftInfo {
+			owner: AccountIdOrCollectionNftTuple::AccountId(owner),
+			royalty,
+			metadata,
+			equipped: false,
+			pending,
+			transferable,
+		};
+
+		Nfts::<T>::insert(collection_id, nft_id, nft);
+
+		// increment nfts counter
+		let nfts_count = collection.nfts_count.checked_add(1).ok_or(ArithmeticError::Overflow)?;
+		Collections::<T>::try_mutate(collection_id, |collection| -> DispatchResult {
+			let collection = collection.as_mut().ok_or(Error::<T>::CollectionUnknown)?;
+			collection.nfts_count = nfts_count;
+			Ok(())
+		})?;
+
+		Ok((collection_id, nft_id))
+	}
+
+	fn nft_mint_directly_to_nft(
+		sender: T::AccountId,
+		owner: (CollectionId, NftId),
+		collection_id: CollectionId,
```

Comment:
isn't `collection_id` redundant now, because it's already part of owner tuple?

Reply:
No, `owner` is the identifier of the NFT that will own the new NFT, `collection_id` is where the NFT should be minted into.   I'd expect we don't want to require these collections be the same.